dbms/src/Storages/DeltaMerge/RowKeyRange.h

```diff
@@ -307,76 +302,48 @@ inline int compare(const RowKeyValueRef & a, const RowKeyValueRef & b)
     }
 }
 
-// TODO (wenxuan): The following compare operators can be simplified using
-// boost::operator, or operator<=> when we upgrade to C++20.
-
-inline int compare(const StringRef & a, const RowKeyValueRef & b)
-{
-    RowKeyValueRef r_a{true, a.data, a.size, 0};
-    return compare(r_a, b);
-}
-
-inline int compare(const RowKeyValueRef & a, const StringRef & b)
-{
-    RowKeyValueRef r_b{true, b.data, b.size, 0};
-    return compare(a, r_b);
-}
-
-inline bool operator<(const RowKeyValueRef & a, const RowKeyValueRef & b)
+inline bool operator==(const RowKeyValueRef & a, const RowKeyValueRef & b)
 {
-    return compare(a, b) < 0;
+    return (a <=> b) == 0;
 }
 
-inline bool operator<=(const RowKeyValueRef & a, const RowKeyValueRef & b)
+inline bool operator!=(const RowKeyValueRef & a, const RowKeyValueRef & b)
 {
-    return compare(a, b) <= 0;
+    return (a <=> b) != 0;
 }
 
-inline bool operator<(const StringRef & a, const RowKeyValueRef & b)
+inline int operator<=>(const StringRef & a, const RowKeyValueRef & b)
 {
-    return compare(a, b) < 0;
+    RowKeyValueRef r_a{true, a.data, a.size, 0};
+    return r_a <=> b;
 }
 
-inline bool operator<(const RowKeyValueRef & a, const StringRef & b)
+inline int operator<=>(const RowKeyValueRef & a, const StringRef & b)
 {
-    return compare(a, b) < 0;
+    RowKeyValueRef r_b{true, b.data, b.size, 0};
+    return a <=> r_b;
 }
 
-inline int compare(Int64 a, const RowKeyValueRef & b)
+inline int operator<=>(Int64 a, const RowKeyValueRef & b)
 {
     RowKeyValueRef r_a{false, nullptr, 0, a};
-    return compare(r_a, b);
+    return r_a <=> b;
 }
 
-inline int compare(const RowKeyValueRef & a, Int64 b)
+inline int operator<=>(const RowKeyValueRef & a, Int64 b)
```

Comment:
amazing

Reply:
😳 Does clang 13 in our CI already support this operator? I though it was only supported since clang 15 or so.